pineappl_py/README.md

```diff
@@ -19,11 +19,19 @@ Run
 
 ```shell
 python -m venv env && . env/bin/activate
-pip install -r requirements.txt
+pip install .[docs]
```

Comment:
That doesn't seem to work.

Reply:
Maybe you need quotes? Ie `pip install '.[docs]'`.

Reply:
No, even `pip install .` doesn't work. Should be fixed with commit 6895357718ef53ccda0fea63d0fa9b87ffcb0329.

Reply:
Mmm this is precisely what RTD is doing: https://readthedocs.org/projects/pineappl/builds/25310692/

Reply:
I'm getting:

```
Processing /home/cschwan/projects/pineappl/pineappl_py                                                                                        
  Installing build dependencies ... done 
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... error                                 
    ERROR: Command errored out with exit status 1:
     command: /home/cschwan/projects/pineappl/pineappl_py/env/bin/python3 /tmp/tmptb0qt4nl_in_process.py prepare_metadata_for_build_wheel /tmp/tmpl83sw3iv
         cwd: /tmp/pip-req-build-55h9jzl7                                                                                                     
    Complete output (13 lines):      
    error: failed to parse manifest at `/tmp/pip-req-build-55h9jzl7/Cargo.toml`
                                                                                                                                              
    Caused by:                      
      error inheriting `version` from workspace root manifest's `workspace.package.version`
                                            
    Caused by:                                                                                                                                
      failed to find a workspace root                                                                                                         
    💥 maturin failed
      Caused by: Cargo metadata failed. Does your crate compile with `cargo build`?
      Caused by: `cargo metadata` exited with an error:                
    Error running maturin: Command '['maturin', 'pep517', 'write-dist-info', '--metadata-directory', '/tmp/pip-modern-metadata-2wy79mhg', '--interpreter', '/home/cschwan/projects/pineappl/pineappl_py/env/bin/python3']' returned non-zero exit status 1.
    Checking for Rust toolchain....      
    Running `maturin pep517 write-dist-info --metadata-directory /tmp/pip-modern-metadata-2wy79mhg --interpreter /home/cschwan/projects/pineappl/pineappl_py/env/bin/python3`
    ----------------------------------------
WARNING: Discarding file:///home/cschwan/projects/pineappl/pineappl_py. Command errored out with exit status 1: /home/cschwan/projects/pineappl/pineappl_py/env/bin/python3 /tmp/tmptb0qt4nl_in_process.py prepare_metadata_for_build_wheel /tmp/tmpl83sw3iv Check the logs for full command
 output.      
ERROR: Command errored out with exit status 1: /home/cschwan/projects/pineappl/pineappl_py/env/bin/python3 /tmp/tmptb0qt4nl_in_process.py prepare_metadata_for_build_wheel /tmp/tmpl83sw3iv Check the logs for full command output.
```

Reply:
yes, I'm getting the same (I already forgot) - I patched the `Cargo.toml` :see_no_evil: ; but so it is not only me ... however this sounds like a maturin problem maybe ... (and RTD can do it)

Reply:
> or actually the manifest definition,

he doesn't like the "inherit from parent package" - if you copy then plain version from above (what my hack is) is works

Reply:
In any case, the fix in the README is correct: if you want to use `maturin develop` you need `pip install maturin` first.

Otherwise, you can use:
```sh
pip install .     # install from source
pip install -e .  # install from source in editable mode
```
and the second is equivalent to `maturin develop`. But it's an alternative, downloading and using Maturin internally. You should choose your workflow.

Reply:
> he doesn't like the "inherit from parent package" - if you copy then plain version from above (what my hack is) is works

And trying with Maturin 1.7?
The version written here is a lower bound, but dating almost two years ago (with a 1.0 release in between...).

Reply:
For me it doesn't - I've always used 1.7.

Reply:
This was caused by an outdated version of `pip`. I updated our README in commit b0de152dfe81591158f25b3b1d6602c150dfc919, which now describes the potential pitfall.